packages/dds/tree/src/index.ts

```diff
@@ -3,243 +3,29 @@
  * Licensed under the MIT License.
  */
 
-export {
-	EmptyKey,
-	type FieldKey,
-	type TreeValue,
-	type FieldMapObject,
-	type NodeData,
-	type GenericTreeNode,
-	type JsonableTree,
-	type GenericFieldsNode,
-	type TreeNodeSchemaIdentifier,
-	type TreeFieldStoredSchema,
-	ValueSchema,
-	TreeNodeStoredSchema,
-	type FieldKindIdentifier,
-	type TreeTypeSet,
-	type TreeStoredSchema,
-	type Revertible,
-	CommitKind,
-	RevertibleStatus,
-	type CommitMetadata,
-	type StoredSchemaCollection,
-	type ErasedTreeNodeSchemaDataFormat,
-	ObjectNodeStoredSchema,
-	MapNodeStoredSchema,
-	LeafNodeStoredSchema,
-	type RevertibleFactory,
-	type RevertibleAlphaFactory,
-	type RevertibleAlpha,
-} from "./core/index.js";
-export { type Brand } from "./util/index.js";
-
-export type {
-	Listeners,
-	IsListener,
-	Listenable,
-	Off,
-} from "@fluidframework/core-interfaces";
+// eslint-disable-next-line no-restricted-syntax
+export * from "./indexCommonApi.js";
```

Comment:
This change does not seem to be described by the PR description. If it needs to be part of this API deprecating event APIs, it would be good to document why this change is being done as part of this in the PR description.

Also it should be documented in the code (here and/or in indexCommonApi) why indexCommonApi exists so people reading the code know what its for and how to maintain it.

Reply:
Note that without knowing why this change is being made, I can't comment on if it's a good idea or done correctly.

Generally, I don't like submodule exports like this since they can very easily break our API reporting / rollup generation tools (since they don't look inside modules like this) and can easily generate errors when packages reexport types from nested modules.

Most of these exports seem to be internal APIs, which I don't see why you would want to modify this way, but if any of them are more public than internal, I think we will leak the internal ones into the rollup with them due to tooling limitations.

Reply:
Also https://github.com/microsoft/FluidFramework/pull/23256 removes many of these exports (they are only use by the dev tool): would be nice to do this change after that one if practical.

Reply:
If the goal here is to avoid `Multiple exports of name 'x'.eslintimport/export`, be aware that that is not a JavaScript or a TypeScript error. That is an eslint error used to detect when people have conflicting exports on accident, letting them know that some of the exports that would otherwise been included in the * export were skipped. Given that this is exactly what you want to happen, I suggest simply suppressing the lint error, and adding a comment about the suppression along the lines of "This reexports all non-conflicting APIs from tree. The event APIs are known to conflict, and this is intended as the exports via core-interfaces are preferred over the deprecated ones from tree".

This should avoid the need to mess with trees's exports (which when undone, fixes the fact that you haven't documented why you had to change them)

Reply:
The refactor was indeed in response to the multiple same export issue. The original thread is https://github.com/microsoft/FluidFramework/pull/23183#discussion_r1854621185 and none of us knew offhand what the specified or most importantly **actual** behavior was for multiple of the same export. Seemed better to avoid the situation altogether. This current pattern appears compatible with our export verifications, and I don't see holes in coverage. Though yes, the pattern should be clearly documented in the code.
Alternatively, we can simplify with lint disables and added comments explaining exactly what happens.

Reply:
Opened a new PR for this change: https://github.com/microsoft/FluidFramework/pull/23313

Reply:
I am going to unresolve this issue because we need to pick one or the other.